### PR TITLE
Architecture overview for Rialto server manager

### PR DESCRIPTION
**Pull request by balasaraswathy-n**

### CODE REVIEW

File: .github/prompts/catalyst-architecture-brief.prompt.md [hunk 1]
```diff
@@ -0,0 +1,807 @@
...
+### Required Sections
+
+1. **Table of Contents** - Include a table of contents for all markdown headings
+2. **Overview** - Overview of the system on who uses this for what
+3. **Problem Definitions & Business Context**
+   - Problem Statement
+   - Primary Users and Use Cases
+   - Non Functional Requirements (availability, performance, security, scalability)
+   - Integration Points with external systems
+4. **C4 System Context Diagram** - Visually appealing system context using Mermaid graph TD format with color-coded categories, emojis, and descriptive connection labels
```
Comment (skywojciechowskim) [inline .github/prompts/catalyst-architecture-brief.prompt.md:130]: The generated diagram contains errors.

File: serverManager/architecture-brief.md [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+## Runtime Behavior
+### Initialization
+1. Host creates service using `create(stateObserver[, config])`.
+2. `ConfigHelper` resolves effective configuration (plus optional JSON file overrides).
+3. `ServiceContext` constructs `SessionServerAppManager` and IPC `Controller`.
+4. `ServerManagerService` preloads session servers if configured.
+
+Detailed initialization behavior from current code paths:
+- When config-file support is enabled, config is read in precedence order: base (`RIALTO_CONFIG_PATH`), SoC (`RIALTO_CONFIG_SOC_PATH`), then overrides (`RIALTO_CONFIG_OVERRIDES_PATH`).
+- `envVariables` from a more important file replace lower-priority values; `extraEnvVariables` are merged last and can overwrite keys.
```
Comment (skywojciechowskim) [inline serverManager/architecture-brief.md]: I think, that it's important to add one thing here.
The values of both envVariables and/or extraEnvVariables from the config files from /etc are wiped, if envVariables and/or extraEnvVariables exist in the overrides file.

File: serverManager/architecture-brief.md [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+## Resource and Socket Contracts
+Sequence details add practical contracts for integrators:
+- Manager-to-session-server control channel uses socketpair point-to-point IPC.
+- App-to-session-server IPC socket may be caller-provided or auto-generated.
+- Display integration may set `WAYLAND_DISPLAY` when display name is provided.
+- Resource descriptor sent in configuration currently models playback-count and web-audio capability, with extension points for richer capability classes (for example HD/UHD tiers).
+- For active sessions, shared memory setup is part of readiness and must be mapped by the app client.
+
+Connection-info contract detail:
+- After transition out of not-running, app manager is expected to query `getAppConnectionInfo()` promptly so the socket name can be passed into container/app launch context.
```
Comment (skywojciechowskim) [inline serverManager/architecture-brief.md:217]: out of Uninitialized. Preloaded servers do not have socket name assigned yet.

File: serverManager/architecture-brief.md [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+| `INACTIVE -> NOT_RUNNING` | `changeSessionServerState(appId, NOT_RUNNING)` | App currently inactive; mapped server exists | Send `SetState(NOT_RUNNING)`; close app IPC; session server emits not-running then self-terminates | Observer receives `stateChanged(appId, NOT_RUNNING)`; manager removes registry/client bookkeeping | If IPC/process termination is abnormal, disconnect/recovery logic may trigger restart/error handling |
+| `ACTIVE -> NOT_RUNNING` | `changeSessionServerState(appId, NOT_RUNNING)` | App active and mapped | Composite cleanup: perform active-resource teardown (as in `ACTIVE -> INACTIVE`) then close app IPC/terminate (as in `INACTIVE -> NOT_RUNNING`) | App ends in not-running with resource cleanup complete; manager updates observer and removes mapping | Failure in either sub-stage may surface as `ERROR` and/or disconnect-triggered recovery path |
+
+### Matrix Notes
+- Observer contract: each completed transition edge should result in manager-to-host `stateChanged(appId, state)` notification.
+- Connection info usage: after leaving not-running, app manager should query `getAppConnectionInfo(appId)` promptly unless socket name was pre-supplied.
```
Comment (skywojciechowskim) [inline serverManager/architecture-brief.md]: After leaving Uninitialized :)

File: serverManager/architecture-brief.md [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+    - Public interfaces are exposed in `public/` using interface-style headers.
+    - Implementation types in `common/`, `service/`, and `ipc/` use PascalCase file names aligned with class names.
+    - Supporting test utility code lives under `serverManagerSim/`.
+- Error handling patterns:
+    - Explicit state-machine error transitions to `ERROR` for failed lifecycle operations.
+    - Timeout/disconnect failures trigger deterministic cleanup (socket unbind/free, process restart/termination path).
+    - Healthcheck and startup timers are treated as first-class failure detectors with policy-driven recovery.
+- Testing approaches:
+    - Unit tests in `tests/unittests/` validate lifecycle logic, configuration behavior, and edge-case handling.
+    - Component/integration tests in `tests/componenttests/` validate multi-process and IPC behavior.
```
Comment (skywojciechowskim) [inline serverManager/architecture-brief.md]: Server manager doesn't have CTs
